### ISSUE TITLE
[BUG] Element stays in whileHover state even on mouseout if it has additional variants configuration for the same property.

### ISSUE THREAD

**Issue by parkerziegler** [framer/motion#1197]
**1. Read the [FAQs](#faqs) 👇**

👀 Read! ✅ 

**2. Describe the bug**

When animating **the same property** via a `staggerChildren` (parent-controlled) and `whileHover` (child-controlled) animation, `framer-motion` persists the `whileHover` state even when the hover is not active. In the example below, we define an initial animation to run on mount that animates `translateY` from 50 to 0, staggering and delaying the children. Additionally, we define a `whileHover` animation on each child that animates `translateY` from 0 to -10 when hover is active.

```tsx
const parentVariants = {
  visible: {
    transition: { staggerChildren: 0.07, delayChildren: 0.2 }
  },
  hidden: {}
};

const childVariants = {
  visible: {
    y: 0,
    opacity: 1,
    transition: {
      y: { stiffness: 1000, velocity: -100 }
    }
  },
  hidden: {
    y: 50,
    opacity: 0,
    transition: {
      y: { stiffness: 1000 }
    }
  }
};

const keys = ["box_1", "box_2", "box_3"];

const App = () => {
  return (
    <div className="App">
      <motion.div
        className="stack-h"
        variants={parentVariants}
        animate="visible"
        initial="hidden"
      >
        {keys.map((key) => (
          <motion.div
            key={key}
            variants={childVariants}
            className="box"
            whileHover={{ x: 10, y: -10, boxShadow: "-0.5rem 0.5rem orange" }}
          />
        ))}
      </motion.div>
    </div>
  );
};
```

On hover, each box animates from `translateY(0px)` to `translateY(-10px)` as expected. When the hover becomes inactive (`mouseout` event fires), the box stays at `translateY(-10px)` rather than animating back to `translateY(0px)`.

I confirmed that if we _just_ have a `whileHover` animation (no additional variants), `whileHover` works as expected and we see the transition back to `translateY(0px)` on mouseout.
 
**3. IMPORTANT: Provide a CodeSandbox reproduction of the bug**

https://codesandbox.io/s/framer-motion-variants-whilehover-bug-ns6dk?file=/src/App.tsx:65-943

**4. Steps to reproduce**

1. Open the above CodeSandbox.
2. Inspect any of the three `steelblue` boxes before hovering. Note that they have `transform: none` applied.
![image](https://user-images.githubusercontent.com/19421190/123323446-5b0dd100-d4ea-11eb-893d-9d53e3105e36.png)
3. Hover over any of the three `steelblue` boxes and then mouseout.
4. Observe that the boxes preserve `translateY(-10px)` even after mouseout. Note that `translateX`, which is not animated in any of the `variants`, animates back to `0px` just fine.
![image](https://user-images.githubusercontent.com/19421190/123323606-97413180-d4ea-11eb-9c05-2da111d7afe8.png)
 

**5. Expected behavior**

I would expect `translateY` in the above example to animate back to `0px` on `mouseout`. This was the behavior in v2 and I suspect may have changed in v3 with the updates to animation priority. However, if animation priority were followed, I would expect that we would animate to the `visible` variant on the child (`translateY(0px)`) on `mouseout`, but instead we stay on the active hover state defined in `whileHover` (`translateY(-10px)`).

**6. Video or screenshots**

![framer-motion-whileHover-bug](https://user-images.githubusercontent.com/19421190/123326422-11bf8080-d4ee-11eb-81d3-5c85ed712109.gif)

**7. Environment details**

macOS 10.14.6
Firefox 89

### `AnimatePresence` isn't working

Have all of its immediate children got a unique `key` prop that **remains the same for that component every render**?

```jsx
// Bad: The index could be given to a different component if the order of items changes
<AnimatePresence>
  {items.map((item, index) => <Component key={index} />)}
</AnimatePresence>
```

```jsx 
// Good: The item ID is unique to each component
<AnimatePresence>
  {items.map((item, index) => <Component key={item.id} />)}
</AnimatePresence>
```

Is the `AnimatePresence` correctly outside of the controlling conditional? `AnimatePresence` must be rendered whenever you expect an `exit` animation to run - it can't do so if it's unmounted!

```jsx
// Bad: AnimatePresence is unmounted - exit animations won't run
{isVisible && (
  <AnimatePresence>
    <Component />
  </AnimatePresence>
)}
```

```jsx
// Good: Only the children are unmounted - exit animations will run
<AnimatePresence>
  {isVisible && <Component />}
</AnimatePresence>
```

**Comment by GMaiolo** [comment]
Note: This is still happening with version 6.3.10

**Comment by mattgperry** [comment]
@GMaiolo  did something in the changelog indicate this may have been fixed in 6.3.10? Otherwise the pinging isn’t necessary.

**Comment by jacobedawson** [comment]
I'm running into this issue as well, element stays in the whileHover state and does not return to the expected initial state.

**Comment by jacobedawson** [comment]
@GMaiolo @parkerziegler it took several frustrating hours, but for whatever reason, I managed to fix this by using keyframes on the problematic property. Combined with having a "wrapper" motion.div I have been able to get a staggerChildren animation and a whileHover gesture that returns to its original state. 

<img width="744" alt="image" src="https://user-images.githubusercontent.com/4788506/211437595-bba716aa-7d26-49c7-aeaa-ac45fc35f781.png">

This may not be the perfect solution, but it's the only one so far that gave me exactly what I want - a staggered entry animation on the children of a container component, alongside an individual hover animation that animated from e.g. opacity 0.5 to opacity 1 and returned to 0.5. 

I'm not sure why the keyframes made a difference here. Having the "wrapper" made sure that the Y transform also returned to 0.
